yoti-sdk-impl/src/test/java/com/yoti/api/client/shareurl/extension/SimpleThirdPartyAttributeExtensionBuilderTest.java

```diff
@@ -0,0 +1,138 @@
+                    .build();
+        } catch (IllegalArgumentException e) {
+            assertThat(e.getMessage(), containsString("definition"));
+            return;
+        }
+        fail("Expected an exception");
+    }
+
+    @Test
+    public void shouldBuildThirdPartyAttributeExtensionWithGivenValues() {
+        Extension<ThirdPartyAttributeContent> extension = new SimpleThirdPartyAttributeExtensionBuilder()
+                .withExpiryDate(SOME_DATE)
+                .withDefinition(SOME_DEFINITION)
+                .build();
+
+        SimpleDateFormat sdf = new SimpleDateFormat(YotiConstants.RFC3339_PATTERN_MILLIS);
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+        String formattedTestDate = sdf.format(SOME_DATE);
+
+        assertEquals(ExtensionConstants.THIRD_PARTY_ATTRIBUTE, extension.getType());
+        assertEquals(formattedTestDate, extension.getContent().getExpiryDate());
+
+        List<AttributeDefinition> definitions = extension.getContent().getDefinitions();
+        assertThat(definitions.size(), is(1));
+        assertThat(definitions.get(0).getName(), is(SOME_DEFINITION));
+    }
+
+    @Test
+    public void shouldBuildThirdPartyAttributeExtensionWithMultipleDefinitions() {
+        List<String> theDefinitions = new ArrayList<>();
+        theDefinitions.add("firstDefinition");
+        theDefinitions.add("secondDefinition");
+
+        Extension<ThirdPartyAttributeContent> extension = new SimpleThirdPartyAttributeExtensionBuilder()
+                .withExpiryDate(SOME_DATE)
+                .withDefinitions(theDefinitions)
+                .build();
+
+        assertEquals(ExtensionConstants.THIRD_PARTY_ATTRIBUTE, extension.getType());
+
+        List<AttributeDefinition> definitions = extension.getContent().getDefinitions();
+        assertThat(definitions.size(), is(2));
+        assertThat(definitions.get(0).getName(), is("firstDefinition"));
+        assertThat(definitions.get(1).getName(), is("secondDefinition"));
+    }
+
+    @Test
+    public void shouldOverwriteSingularlyAddedDefinition() {
+        List<String> theDefinitions = new ArrayList<>();
+        theDefinitions.add("firstDefinition");
+        theDefinitions.add("secondDefinition");
+
+        Extension<ThirdPartyAttributeContent> extension = new SimpleThirdPartyAttributeExtensionBuilder()
+                .withExpiryDate(SOME_DATE)
+                .withDefinition(SOME_DEFINITION)
+                .withDefinitions(theDefinitions)
+                .build();
+
+        assertEquals(ExtensionConstants.THIRD_PARTY_ATTRIBUTE, extension.getType());
+
+        List<AttributeDefinition> definitions = extension.getContent().getDefinitions();
+        assertThat(definitions.size(), is(2));
+        assertThat(definitions.get(0).getName(), is("firstDefinition"));
+        assertThat(definitions.get(1).getName(), is("secondDefinition"));
+    }
+
+    @Test
+    public void doTest() throws Exception {
```

Comment:
I think this test maybe isn't finished, or can be removed 👍

Reply:
Fixed in https://github.com/getyoti/yoti-java-sdk/pull/142/commits/04fe75becb7960203ed6e759fb81251c010bf64d